src/test/kotlin/org/wfanet/panelmatch/common/certificates/gcloud/CertificateAuthorityTest.kt

```diff
@@ -0,0 +1,175 @@
+import com.google.cloud.security.privateca.v1.X509Parameters.CaOptions
+import com.google.common.base.Predicates.instanceOf
+import com.google.common.truth.Truth.assertThat
+import com.google.protobuf.Duration
+import com.google.protobuf.kotlin.toByteString
+import org.wfanet.measurement.common.crypto.testing.FIXED_CA_CERT_PEM_FILE
+
+private val CONTEXT =
+  CertificateAuthority.Context(
+    organization = "some-organization",
+    commonName = "some-common-name",
+    hostname = "example.com",
+    validDays = 5
+  )
+
+private val ROOT_X509 by lazy { readCertificate(FIXED_CA_CERT_PEM_FILE) }
+private val ROOT_PUBLIC_KEY by lazy { ROOT_X509.publicKey }
+
+// Set the X.509 fields required for the certificate.
+private val X509PARAMETERS = X509Parameters.newBuilder()
+  .setKeyUsage(
+    KeyUsage.newBuilder()
+      .setBaseKeyUsage(
+        KeyUsageOptions.newBuilder()
+          .setDigitalSignature(true)
+          .setKeyEncipherment(true)
+          .setCertSign(true)
+          .build()
+      )
+      .setExtendedKeyUsage(
+        ExtendedKeyUsageOptions.newBuilder().setServerAuth(true).build()
+      )
+      .build()
+  )
+  .setCaOptions(CaOptions.newBuilder().setIsCa(true).buildPartial())
+  .build()
+
+private val DURATION: Duration =
+  Duration.newBuilder().setSeconds(CONTEXT.validDays.toLong() * 86400).build()
+
+private val CREATECERTIFICATECLIENT: CreateCertificateClient = mock<PrivateCaClient>()
+
+@RunWith(JUnit4::class)
+class CertificateAuthorityTest {
+
+  @Test
+  fun mockGenerateX509CertificateAndPrivateKeyTest() {
+
+    val certificateAuthority =
+      CertificateAuthority(
+        "some-project-id",
+        "some-ca-location",
+        "some-pool-id",
+        "some-certificate-authority-name",
+        "some-certificate-name" ,
+        "some-common-name",
+        "some-org-name",
+        "some-domain-name",
+        DURATION,
+        CREATECERTIFICATECLIENT,
+        X509PARAMETERS
+      )
+
+    val (x509, privateKey) =
+      runBlocking { certificateAuthority.generateX509CertificateAndPrivateKey() }
+
+    assertThat(instanceOf(privateKey::class.java)).isEqualTo(instanceOf(PrivateKey::class.java))
+    assertThat(instanceOf(x509::class.java)).isEqualTo(instanceOf(X509Certificate::class.java))
+  }
+
+  @Test
+  suspend fun SpecificCertificateAuthorityRequestTest() {
+
+    val createCertificateRequest: CreateCertificateRequest =
+      generateCreateCertificateRequest("astute-smile-334323",
+        "us-east4",
+        "mahi-ca-test-1-common-name",
+        "20211205-asq-atj",
+        "220211205-ny6-r5m" ,
+        "wfa-mahi-test-ca-name",
+        "wfa-mahi-test",
+        "TODO - UNSURE", DURATION)
+
+    val response = CREATECERTIFICATECLIENT.createCertificate(createCertificateRequest)
+    val x509Certificate = readCertificate(response.pemCertificate.byteInputStream())
+
+    assertThat(instanceOf(x509Certificate::class.java)).isEqualTo(instanceOf(X509Certificate::class.java))
```

Comment:
@efoxepstein This doesn't seem right regarding the mock test. Is there another way to compare the types? (same as above)? Should I even keep this test since it has specific params?